test/cluster_test.go

```diff
@@ -89,6 +89,22 @@ func checkSubInterest(t *testing.T, s *server.Server, accName, subject string, t
 	})
 }
 
+func checkNoSubInterest(t *testing.T, s *server.Server, accName, subject string, timeout time.Duration) {
+	t.Helper()
+	acc, err := s.LookupAccount(accName)
+	if err != nil {
+		t.Fatalf("error looking up account %q: %v", accName, err)
+	}
+
+	start := time.Now()
```

Comment:
You need to use checkFor() because the way this is written below, this helper will fail (with t.Fatalf) before the allowed time. Meaning that as soon as it finds subscription interest, the test will fail, even if the interest would have disappeared before `timeout`. So it should be:
```
checkFor(t, timeout, 5*time.Millisecond, func() error {
    if acc.SubscriptionInterest(subject) {
        return fmt.Errorf("did not expect interest for %q", subject)
    }
    return nil
})
```

Reply:
I am doing the opposite and want to wait to see if it shows up.

Reply:
What? You added this new helper called `checkNoSubInterest` and it is failing the test if there is an interest (but my comment is that it is failing too soon). So I do believe that this helper should fail if there is subscription interest, but only after waiting for the max `timeout`.
If you want to check for sub interest, there is already an existing helper called `checkSubInterest` defined just before this one that you added.

Reply:
I want to check that interest for a certain subject does not show up in a given timeframe.

Reply:
So you mean that you know that there is no subject interest and don't want for it to show up at all, not that the subject interest may exist but you expect it to disappear before a given timeout.
Ok, I understand now, but this is confusing since most (if not all) of our helpers are waiting for a condition to happen. For instance the one that exists just above the one you added checks that there will be subscription interest (waits for subject propagation basically).

Reply:
Yes this one is opposite, want to make sure interest does not propagate.